overlord/devicestate/devicemgr.go

```diff
@@ -99,6 +99,9 @@ type DeviceManager struct {
 	bootRevisionsUpdated bool
 
 	seedTimings *timings.Timings
+	// these are used as needed as cache during StartUp and cleared after
+	earlyDeviceCtx  snapstate.DeviceContext
+	earlyDeviceSeed seed.Seed
```

Comment:
Maybe these could be part of an `earlyDeviceInfo` struct, which we store as a pointer here and nullify once it's no longer needed?

Reply:
the refactoring I'm discussing elsewhere would make these have different lifetimes so I wouldn't bother, The seed would need to be kept around outside of StartUp